server/bin/pbench-sync-satellite.sh

```diff
@@ -266,6 +266,10 @@ if [ -s ${state_change_log} ] ;then
     fi
 fi
 
+if [[ ${nerrs} == 0 ]] ; then
+    find "${logdir_for_remote}"/* -name "*-UTC" -type d -print0|xargs -0 rm -r --
```

Comment:
This does not seem right still.

The main loop for the satellite is at [line #148](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R148) where it loops over the list of hosts (really, pbench "`controller`" names), and executes a sequence of tasks.

At [line #173](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R173), we create a directory in the `${logdir}` (which is just `${logdir_for_remote}/${TS}`) for the given `host`, and we proceed to create 3 (three) files in that directory:

 1. `md5-checks.log`, @ [line #200](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R200)
 2. `ok-checks.log`, @ [line #203](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R203)
 3. `fail-checks.log`, # [line #217](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R217)

Then we can see that at [line #249](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R249) we essentially copy the contents of `ok-checks.log` into the `${state_change_log}` file.

So we should be able to do the following to clean things up better:

 1. Always remove the `ok-checks.log` file at the end of each iteration
 2. Remove the `fail-checks.log` file if it is empty
 3. Add a third `grep` of the `md5-checks.log` file to look for all the lines NOT containing `OK` or `FAILED`, and removing `md5-checks.log` if there were no matching lines
 4. Add an `rmdir ${logdir}/${host}` which will remove the directory for the given host if there were no errors

Can you try that out instead?

Reply:
Ok.. sure